### ISSUE TITLE
Possibility to detect a command is not runnable before execution

### ISSUE THREAD

**Issue by ekkinox** [ekkinox/yai#17]
For example : 

![Screenshot 2023-04-18 at 21 15 05 (1)](https://user-images.githubusercontent.com/6922385/232882468-7dc3ad47-6ccf-425f-8eb2-eac41509f5cf.png)

Yo could detect that the proposed command involves software not available, and propose remediation for this.

Thx @Skyld for the suggestion.

**Comment by mateothegreat** [comment]
Then make another prediction about how to install it.. maybe have side effects support?

**Comment by ekkinox** [comment]
> Then make another prediction about how to install it.. maybe have side effects support?

yeah need to check how to make this happen, especially if proposed command line contains more than 1 programs that are not installed :D

The workaround I use for now is:
- me: do some stuff
- yo: propose a command with a binary I dont have
- me: try to run and fail / cancal
- me: ask to to install missing stuff

not optimal, but works
